core/src/main/java/tech/tablesaw/columns/strings/StringMapFunctions.java

```diff
@@ -267,28 +312,47 @@ default DoubleColumn distance(Column<?> column2) {
    * @param columns the column to append
    * @return the new column
    */
-  default StringColumn join(String separator, Column<?>... columns) {
+  default StringColumn join(String separator, Column... columns) {
     StringColumn newColumn = StringColumn.create(name() + "[column appended]", this.size());
     for (int r = 0; r < size(); r++) {
-      String result = getString(r);
+      StringBuilder result = new StringBuilder(getString(r));
       for (Column<?> stringColumn : columns) {
-        result = result + separator + stringColumn.get(r);
+        result.append(separator).append(stringColumn.get(r));
       }
-      newColumn.set(r, result);
+      newColumn.set(r, result.toString());
     }
     return newColumn;
   }
 
   /**
    * Return a copy of this column with the given string appended to each element
    *
-   * @param append the string to append
+   * @param stringsToAppend the stringified objects to append
+   * @return the new column
+   */
+  default StringColumn concatenate(Object... stringsToAppend) {
+    StringColumn newColumn = StringColumn.create(name() + "[append]", this.size());
+    for (int r = 0; r < size(); r++) {
+      for (Object o : stringsToAppend) {
+        newColumn.set(r, getString(r) + o);
+      }
+    }
+    return newColumn;
+  }
+
+  /**
+   * Return a copy of this column with the corresponding value of each column argument appended to
+   * each element. getString is used to ensure the value returned by the args are strings
+   *
+   * @param stringColumns the string columns to append
    * @return the new column
    */
-  default StringColumn concatenate(String append) {
+  default StringColumn concatenate(Column... stringColumns) {
```

Comment:
Should this take a `StringColumn` or is that too specific? Otherwise maybe this should be `Column<?>...`

I'm getting compilation warnings on lines 315, 350, and 354 due to missing generics